File: Equilibrium Point[23-September-2023]/Solution.cpp
//  >> CPP CODE <<

class Solution{
    public:
    int equilibriumPoint(long long a[], int n) {
    
        long long sumArray=0;
        
        for(int i=0;i<n;i++) sumArray+=a[i];
        
        long long ss=0;
        
        for(int i=0;i<n;i++)
        {
            if(ss==sumArray-ss-a[i]) return i+1;
            ss+=a[i];
        }
        
            return -1;
    }

};

File: Equilibrium Point[23-September-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    # Complete this function
    
    #Function to find equilibrium point in the array.
    def equilibriumPoint(self,A, N):
        sumArray=sum(A);
        
        ss=0
        
        for i in range(N):
            if(ss==sumArray-ss-A[i]): return i+1;
            
            ss+=A[i]
        
        return -1
